packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -442,14 +444,14 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ReinitializableBase
     )
         public
     {
-        // Cannot finalize withdrawal with value when custom gas token mode is enabled.
-        if (_isUsingCustomGasToken()) {
-            if (_tx.value > 0) revert OptimismPortal_NotAllowedOnCGTMode();
-        }
-
         // Cannot finalize withdrawal transactions while the system is paused.
         _assertNotPaused();
 
+        // Cannot finalize withdrawal with value when custom gas token mode is enabled.
+        if (_isInvalidCGTWithdrawal(_tx.value)) {
```

Comment:
I liked the original check structure slightly better because it mirrors how the `_isUsingLockbox` check looks